### PR TITLE
Search for run names.

### LINKED ISSUE #733: search for run names [27h]
In codechecker 5.9 it is not possible to filter run names in the run view.

Add a text field for the run view to filter run names.
The text field should support auto-complete.

Implementation:

- [x] add new text field to GUI: 2h
- [x] add new api function to query available run names based on text fragment (Auto-complete): 6h
- [x] add new parameter to the getRunData() function to filter the rundatalist: 3h
- [x] complete GUI actions: 8h
- [x] implemnet test cases: 8h
-----
27h

### PR DESCRIPTION
**Pull request by csordasmarton**
Fixes #733

### CODE REVIEW

File: libcodechecker/server/client_db_access_handler.py [hunk 1]
```diff
@@ -272,8 +272,12 @@ def getRunData(self):
                 .group_by(Report.run_id) \
                 .subquery()
 
-            q = session.query(Run, stmt.c.report_count) \
-                .outerjoin(stmt, Run.id == stmt.c.run_id) \
+            q = session.query(Run, stmt.c.report_count)
+
+            if run_name_filter is not None:
+                q = q.filter(Run.name.like(run_name_filter + '%'))
```
Comment (Xazax-hun) [inline libcodechecker/server/client_db_access_handler.py]: So we are doing a prefix search?

Comment (whisperity) [inline libcodechecker/server/client_db_access_handler.py]: Maybe we shan't. :slightly_frowning_face: Infix searching could be more appropriate, especially for the naming conventions used internally. :/

File: www/scripts/codecheckerviewer/ListOfRuns.js [hunk 2]
```diff
@@ -285,10 +323,38 @@ function (declare, domConstruct, ItemFileWriteStore, topic, Dialog, Button,
     }
   });
 
+  var RunFilter = declare(ContentPane, {
+    constructor : function () {
+      var that = this;
+
+      this._runFilter = new TextBox({
+        id          : 'runs-filter',
+        placeHolder : 'Run name filter...',
```
Comment (whisperity) [inline www/scripts/codecheckerviewer/ListOfRuns.js]: Maybe a better placeholder would be `Search for runs...`?

File: tests/functional/report_viewer_api/test_run_filter.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    _ccClient = None
+
+    def setUp(self):
+        test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + test_workspace)
+
+        # Get the clang version which is tested.
+        self._clang_to_test = env.clang_to_test()
```
Comment (gyorb) [inline tests/functional/report_viewer_api/test_run_filter.py]: Unused variables in the tests could be cleaned up.

**Pull request merged**